src/DataGrid.php

```diff
@@ -1624,12 +1626,14 @@ public function createComponentFilter()
 		 */
 		$form->addSelect('per_page', '', $this->getItemsPerPageList())
 			->setTranslator(null);
+		$validationscope[] = $form['per_page'];
 
 		if (!$form->isSubmitted()) {
 			$form['per_page']->setValue($this->getPerPage());
 		}
 
-		$form->addSubmit('per_page_submit', 'ublaboo_datagrid.per_page_submit');
+		$form->addSubmit('per_page_submit', 'ublaboo_datagrid.per_page_submit')
+			->setValidationScope($validationscope);
```

Comment:
Hi, you need to flatten `$validationscope` before setting it. You can use `Arrays::flatten` from `Nette\Utils`.